### DISCUSSION TITLE
`no-shadow`, based on `https://eslint.org/docs/latest/rules/no-shadow`

### DISCUSSION THREAD

**Discussion by jpike88** [biomejs/biome#535] [Ideas]
Just had a nasty bug arise because a refactor caused this scenario to happen:

```
let x;

// blah blah

if(y){
     const x = somethingUnrelated;
}

// crap... x was not set in outer scope!!!

```

Found the above rule that would prevent this kind of issue occurring again, I believe very valuable to not be reusing variable names like that inside inner scope.

**Comment by Conaclos** [comment]
This corresponds to eslint [no-shadow](https://eslint.org/docs/latest/rules/no-shadow) rule.

**Comment by ematipico** [comment]
Thank you @jpike88 , I suggest copying the message at #3 , that's where suggest new rules from other sources (if they aren't there yet).
